chroma-manager/tests/integration/shared_storage_configuration/test_power_control.py

```diff
@@ -9,10 +9,16 @@ class ChromaPowerControlTestCase(ChromaIntegrationTestCase):
 
     def setUp(self):
         super(ChromaPowerControlTestCase, self).setUp()
+        
+        # Even though the tests only need 1 server, we need to add a server
+        # and its HA peer in order to ensure that the peer doesn't send
+        # outdated CIB data over. There is an assumption here that the
+        # servers listed in the configuration are ordered by HA peer groups.
+        servers = config['lustre_servers'][0:2]
```

Comment:
Why the switch form `self.TEST_SERVERS` to using `config['lustre_servers']` directly rather than redefining `self.TEST_SERVERS`?

I have a feeling that inconsistent use (or lack thereof) of `self.TEST_SERVERS` is part of our inconsistent environment problems.

Reply:
> I have a feeling that inconsistent use (or lack thereof) of self.TEST_SERVERS is part of our inconsistent environment problems.

Redefining: `self.TEST_SERVERS` is what causes inconsistent setup and teardown. By not redefining it explicitly, we ensure we do the full setup / teardown.

Reply:
> Redefining: self.TEST_SERVERS is what causes inconsistent setup and teardown. By not redefining it explicitly, we ensure we do the full setup / teardown.

Well, only because we defer the actions of tearing down until the next test's setUp() by which time TEST_SERVERS has been redefined to be for that next test.  Maybe we just need to bite off on cleaning up after tests actually after the test runs rather than waiting until the next test starts to do the clean up.

Reply:
> Maybe we just need to bite off on cleaning up after tests actually after the test runs rather than waiting until the next test starts to do the clean up.

We should do teardown operations in teardown, yes. But again, the goal here is to get these tests operating reliably, it's not to refactor more of the test framework than I need to.

The work to refactor more of the test framework should happen in: https://github.com/intel-hpdd/intel-manager-for-lustre/issues/327